civis/base.py

```diff
@@ -220,10 +224,27 @@ def _civis_state(self):
     @property
     def _state(self):
         """State of the CivisAsyncResultBase in `future` language."""
+        if self.__state is not None:
+            return self.__state
```

Comment:
should we have some check here that is validating that `self._state` and `self.__state` are in sync?

Reply:
The issue is that we don't want them to be in sync while the `concurrent.futures.Future.set_result()` and `set_exception()` calls are happening. At that point, the platform job has finished, so (before this PR) `CivisFuture.state` would return `futures._base.FINISHED` when python expects it to return `futures._base.RUNNING`.